superdesk/eve_backend.py

```diff
@@ -189,26 +203,29 @@ def delete(self, endpoint_name, lookup):
         search_backend = self._lookup_backend(endpoint_name)
         docs = self.get_from_mongo(endpoint_name, lookup=lookup, req=ParsedRequest())
         ids = [doc[config.ID_FIELD] for doc in docs]
-        backend.remove(endpoint_name, {config.ID_FIELD: {'$in': ids}})
+        removed_ids = ids
+        logger.info("total documents to be removed {}".format(len(ids)))
         if search_backend and ids:
-            self._remove_documents_from_search_backend(endpoint_name, ids)
+            removed_ids = []
+            # first remove it from search backend, so it won't show up. when this is done - remove it from mongo
+            for _id in ids:
+                try:
+                    self.remove_from_search(endpoint_name, _id)
+                    removed_ids.append(_id)
+                except:
+                    continue
```

Comment:
It would be good to log the remove_from_search error to see which items cannot be removed. There are few other places this function is called so maybe it would be better within the function.